### PR TITLE
[Hotfix][OptionRule] Fix option rule about all connectors

### PR DESCRIPTION
**Pull request by EricJoy2048**

### CODE REVIEW

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java [hunk 1]
```diff
@@ -146,43 +153,49 @@ public Builder required(Option<?>... options) {
...
             this.requiredOptions.add(exclusiveRequiredOptions);
             return this;
         }
 
-        /**
-         * Conditional options, These options are required if the {@link Option} == expectValue.
-         */
-        public <T> Builder conditional(Option<T> option, T expectValue, Option<?>... requiredOptions) {
-            return conditional(Condition.of(option, expectValue), requiredOptions);
-        }
```
Comment (ashulin) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java:170]: IMO, The tool method of a single expected value need not be deleted.

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java [hunk 2]
```diff
@@ -146,43 +153,49 @@ public Builder required(Option<?>... options) {
...
         }
 
-        public Builder exclusive(RequiredOption.ExclusiveRequiredOptions exclusiveRequiredOptions) {
+        public Builder exclusive(@NonNull RequiredOption.ExclusiveRequiredOptions exclusiveRequiredOptions) {
+            exclusiveRequiredOptions.getExclusiveOptions().forEach(option -> {
+                verifyDuplicate(option, "ExclusiveOption");
+            });
             this.requiredOptions.add(exclusiveRequiredOptions);
             return this;
         }
```
Comment (ashulin) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java]: IMO, delete the method.

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java (hunk 1 above)
Comment (EricJoy2048) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java:170]: > IMO, The tool method of a single expected value need not be deleted.

Ok.

**Review by TaoZex — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**